Mlem/App/Views/Shared/ErrorView.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  Error View.swift
+//  Mlem
+//
+//  Created by David Bureš on 19.06.2022.
+//
+
+import Combine
+import MlemMiddleware
+import SwiftUI
+import UniformTypeIdentifiers
+
+struct ErrorView: View {
+    @AppStorage("developerMode") var developerMode: Bool = false
+    
+    @State var errorDetails: ErrorDetails
+    
+    @State private var showingFullError: Bool = false
+    @State private var refreshInProgress: Bool = false
+    
+    var timer = Timer.publish(every: 1, tolerance: 0.5, on: .main, in: .common)
+        .autoconnect()
+    
+    init(_ errorDetails: ErrorDetails) {
+        var errorDetails = errorDetails
+        _errorDetails = State(wrappedValue: errorDetails)
+    }
+
+    var body: some View {
+        VStack(spacing: 15) {
+            if showingFullError {
+                if let error = errorDetails.error {
+                    if let error = error as? ApiClientError {
+                        errorDetails(error.description)
+                    } else {
+                        errorDetails(error.localizedDescription)
+                    }
+                }
+            } else {
+                if let systemImage = errorDetails.systemImage {
+                    Image(systemName: systemImage)
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(height: 50)
+                }
+                Text(errorDetails.title ?? "Something went wrong.")
+                    .font(.title3.bold())
+                    .foregroundStyle(.primary)
```

Comment:
(requires palette import)

```suggestion
                    .foregroundStyle(palette.primary)
```